[2024-10-31T20:38:54.899+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-10-31T20:38:54.950+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: sensor_data_consumer.sensor_data_consumer manual__2024-10-31T20:38:50.079425+00:00 [queued]>
[2024-10-31T20:38:54.967+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: sensor_data_consumer.sensor_data_consumer manual__2024-10-31T20:38:50.079425+00:00 [queued]>
[2024-10-31T20:38:54.968+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2024-10-31T20:38:55.035+0000] {taskinstance.py:2330} INFO - Executing <Task(SparkSubmitOperator): sensor_data_consumer> on 2024-10-31 20:38:50.079425+00:00
[2024-10-31T20:38:55.059+0000] {standard_task_runner.py:64} INFO - Started process 260 to run task
[2024-10-31T20:38:55.065+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'sensor_data_consumer', 'sensor_data_consumer', 'manual__2024-10-31T20:38:50.079425+00:00', '--job-id', '41', '--raw', '--subdir', 'DAGS_FOLDER/***_consumer.py', '--cfg-path', '/tmp/tmpl0p_0ly1']
[2024-10-31T20:38:55.071+0000] {standard_task_runner.py:91} INFO - Job 41: Subtask sensor_data_consumer
[2024-10-31T20:38:55.182+0000] {task_command.py:426} INFO - Running <TaskInstance: sensor_data_consumer.sensor_data_consumer manual__2024-10-31T20:38:50.079425+00:00 [running]> on host a9c05964d4b3
[2024-10-31T20:38:55.494+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Bich Ly' AIRFLOW_CTX_DAG_ID='sensor_data_consumer' AIRFLOW_CTX_TASK_ID='sensor_data_consumer' AIRFLOW_CTX_EXECUTION_DATE='2024-10-31T20:38:50.079425+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-10-31T20:38:50.079425+00:00'
[2024-10-31T20:38:55.495+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-10-31T20:38:55.561+0000] {base.py:84} INFO - Using connection ID 'spark_default' for task execution.
[2024-10-31T20:38:55.562+0000] {spark_submit.py:344} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --name KafkaSparkHDFS spark_streaming_job.py
[2024-10-31T20:39:13.629+0000] {spark_submit.py:495} INFO - python3: can't open file '/opt/***/spark_streaming_job.py': [Errno 2] No such file or directory
[2024-10-31T20:39:13.868+0000] {spark_submit.py:495} INFO - 24/10/31 20:39:13 INFO ShutdownHookManager: Shutdown hook called
[2024-10-31T20:39:13.917+0000] {spark_submit.py:495} INFO - 24/10/31 20:39:13 INFO ShutdownHookManager: Deleting directory /tmp/spark-4e1a534b-52ad-4806-bb7e-a69cfc842e83
[2024-10-31T20:39:14.236+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-10-31T20:39:14.327+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 157, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 426, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --name KafkaSparkHDFS spark_streaming_job.py. Error code is: 2.
[2024-10-31T20:39:14.367+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=sensor_data_consumer, task_id=sensor_data_consumer, run_id=manual__2024-10-31T20:38:50.079425+00:00, execution_date=20241031T203850, start_date=20241031T203854, end_date=20241031T203914
[2024-10-31T20:39:14.437+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 41 for task sensor_data_consumer (Cannot execute: spark-submit --master spark://spark-master:7077 --name KafkaSparkHDFS spark_streaming_job.py. Error code is: 2.; 260)
[2024-10-31T20:39:14.463+0000] {local_task_job_runner.py:243} INFO - Task exited with return code 1
[2024-10-31T20:39:14.548+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-10-31T20:39:14.573+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
